bindings/python/RobotInterface.cpp

```diff
@@ -0,0 +1,106 @@
+{
+    namespace py = ::pybind11;
+    using namespace yarp::dev;
+    py::class_<PolyDriver, std::shared_ptr<PolyDriver>>(module, "PolyDriver");
+}
+
+void BipedalLocomotion::bindings::CreatePolyDriverDescriptor(pybind11::module& module)
+{
+    namespace py = ::pybind11;
+    using namespace BipedalLocomotion::RobotInterface;
+    using namespace BipedalLocomotion::ParametersHandler;
+
+    py::class_<PolyDriverDescriptor>(module, "PolyDriverDescriptor")
+        .def(py::init())
+        .def_readwrite("key", &PolyDriverDescriptor::key)
+        .def_readwrite("poly", &PolyDriverDescriptor::poly)
+        .def("is_valid", &PolyDriverDescriptor::isValid);
+
+    module.def(
+        "construct_remote_control_board_remapper",
+        [](std::shared_ptr<IParametersHandler> handler) -> PolyDriverDescriptor {
+            return constructRemoteControlBoardRemapper(handler);
+        },
+        py::arg("handler"));
+
+    module.def(
+        "construct_generic_sensor_client",
+        [](std::shared_ptr<IParametersHandler> handler) -> PolyDriverDescriptor {
+            return constructGenericSensorClient(handler);
+        },
+        py::arg("handler"));
+}
+
+void BipedalLocomotion::bindings::CreateIRobotControl(pybind11::module& module)
+{
+    namespace py = ::pybind11;
+    using namespace BipedalLocomotion::RobotInterface;
+
+    py::class_<IRobotControl> iRobotControl(module, "IRobotControl");
+    py::enum_<IRobotControl::ControlMode>(iRobotControl, "ControlMode")
+        .value("Position", IRobotControl::ControlMode::Position)
+        .value("PositionDirect", IRobotControl::ControlMode::PositionDirect)
+        .value("Velocity", IRobotControl::ControlMode::Velocity)
+        .value("Torque", IRobotControl::ControlMode::Torque)
+        .value("Idle", IRobotControl::ControlMode::Idle)
+        .value("Unknown", IRobotControl::ControlMode::Unknown)
+        .export_values();
+}
+
+void BipedalLocomotion::bindings::CreateYarpRobotControl(pybind11::module& module)
+{
+    namespace py = ::pybind11;
+    using namespace BipedalLocomotion::ParametersHandler;
+    using namespace BipedalLocomotion::RobotInterface;
+
+    py::class_<YarpRobotControl, IRobotControl>(module, "YarpRobotControl")
+        .def(py::init())
+        .def(
+            "initialize",
+            [](YarpRobotControl& impl, std::shared_ptr<IParametersHandler> handler) -> bool {
+                return impl.initialize(handler);
+            },
+            py::arg("handler"))
+        .def("set_driver", &YarpRobotControl::setDriver, py::arg("driver"))
+        .def("check_motion_done",
+             [](YarpRobotControl& impl) {
+                 bool motionDone{false};
+                 bool isTimeExpired{false};
+                 std::vector<std::pair<std::string, double>> info;
+                 bool isOk = impl.checkMotionDone(motionDone, isTimeExpired, info);
+
+                 return std::make_tuple(isOk, motionDone, isTimeExpired, info);
```

Comment:
This seems a complex output, did you double check in particular that `info` is mapped to native Python types?

Reply:
Just checked if I print `info` it contains data of the form 
```
[('r_hip_pitch', 0.5426321143221097), ('r_ankle_roll', 0.05311822268293803)]
```

Reply:
Ok great!